src/nupic/regions/TMRegion.cpp

```diff
@@ -248,17 +247,19 @@ void TMRegion::compute() {
   Output *out;
   out = getOutput("bottomUpOut");
   if (out && (out->hasOutgoingLinks() || LogItem::isDebug())) {
-    auto active = tm_->getActiveCells();         // sparse
-    auto predictive = tm_->getPredictiveCells(); // sparse
-    if (args_.orColumnOutputs) {
-      // aggregate to columns
-      active = VectorHelpers::sparse_cellsToColumns<CellIdx>(active, args_.cellsPerColumn);
-      predictive = VectorHelpers::sparse_cellsToColumns<CellIdx>(predictive, args_.cellsPerColumn);
-    }
     SDR& sdr = out->getData().getSDR();
-    VectorHelpers::unionOfVectors<CellIdx>(sdr.getSparse(), active, predictive);
-    sdr.setSparse(sdr.getSparse()); // to update the cache in SDR.
-
+    if (args_.orColumnOutputs) { //aggregate to columns
```

Comment:
OT: but I think TM's bottomUpOutput should always be only "as columns". 
And we have "activeCells" output, + issue to allow "predictiveCells"